### PR TITLE
Portal Two update and bug fixes

### PR DESCRIPTION
**Pull request by sjpadgett**
- Revamped Patient Secure Chat
- Bug fix online payment where send receipt not working.
- Signature in documents & dashboard fixes(support admin signature now in audits)

### CODE REVIEW

**Comment by sjpadgett** [comment]
@bradymiller , Can I pull down 5.0.1 branch from master and commit these changes(with dependencies for the 5.0.1 tree) so that portal can maintain backwards compatibility? Need to patch these fixes (and others) somehow...

File: portal/portal_payment.php [hunk 1]
```diff
@@ -343,24 +354,38 @@ function calcTaxes($row, $amount)
 
 <title><?php echo xlt('Receipt for Payment'); ?></title>
 
-<script type="text/javascript" src="<?php echo $GLOBALS['assets_static_relative']; ?>/jquery-1-11-3/dist/jquery.js"></script>
+<script type="text/javascript" src="<?php echo $GLOBALS['assets_static_relative']; ?>/jquery-min-1-11-3/index.js"></script>
```
Comment (bradymiller) [inline portal/portal_payment.php]: The original path was fine here??

File: portal/portal_payment.php [hunk 2]
```diff
@@ -441,7 +466,7 @@ function notifyPatient(){
     font-weight: normal
 }
 </style>
-<script type="text/javascript" src="<?php echo $GLOBALS['assets_static_relative']; ?>/jquery-creditcardvalidator/jquery.creditCardValidator.js"></script>
+<script type="text/javascript" src="<?php echo $GLOBALS['assets_static_relative']; ?>/jquery-creditcardvalidator-1-1-0/jquery.creditCardValidator.js"></script>
```
Comment (bradymiller) [inline portal/portal_payment.php]: need to revert this line back

File: portal/messaging/secure_chat.php [hunk 3]
```diff
@@ -492,22 +507,22 @@ public function pingAction()
...
 
-    <script type='text/javascript' src="<?php echo $GLOBALS['assets_static_relative']; ?>/angular/angular.min.js"></script>
-    <script type='text/javascript' src="<?php echo $GLOBALS['assets_static_relative']; ?>/angular-summernote/dist/angular-summernote.js"></script>
-     <script type='text/javascript' src="<?php echo $GLOBALS['assets_static_relative']; ?>/angular-sanitize/angular-sanitize.min.js"></script>
-    <script src='<?php echo $GLOBALS['assets_static_relative']; ?>/checklist-model/checklist-model.js'></script>
+    <script type='text/javascript' src="<?php echo $GLOBALS['assets_static_relative']; ?>/angular-1-5-8/angular.min.js"></script>
+    <script type='text/javascript' src="<?php echo $GLOBALS['assets_static_relative']; ?>/angular-summernote-0-8-1/dist/angular-summernote.js"></script>
+     <script type='text/javascript' src="<?php echo $GLOBALS['assets_static_relative']; ?>/angular-sanitize-1-5-8/angular-sanitize.min.js"></script>
+    <script src='<?php echo $GLOBALS['assets_static_relative']; ?>/checklist-model-0-10-0/checklist-model.js'></script>
```
Comment (bradymiller) [inline portal/messaging/secure_chat.php]: All these asset directory changes need to be reverted.

**Comment by bradymiller** [comment]
hi @sjpadgett ,
After the assets directory changes are all addressed and this is in master, I am guessing this won't be too tough to bring into 5.0.1 (since you won't be touching those paths in the commit, there shouldn't be any merge issues).
Thanks! -brady

**Comment by sjpadgett** [comment]
I discarded those chunks so how they made it up I don't know and to be honest this is not worth keeping up with. Withdrawing PR.

**Comment by bradymiller** [comment]
@sjpadgett ,
Don't let composer/npm take you down :)
If needed, I'd be glad to fix those paths and get this into the codebase.
-brady

**Comment by sjpadgett** [comment]
Composer/npm has nothing to do with it. This was a git screw up. I'm not in the mood to maintain two version of Portal anyway so i'll offer updates to 5.0.1 tree so people can get if they want.

**Comment by sjpadgett** [comment]
@bradymiller looks like it was a Jerry error and not git. worked first commit but failed travis and I forgot to discard the offending hunks again before commit to fix travis error. So maintaining this for 5.0.1/5.0.2 is not so bad if I would stop being an air head... Please look mess over to check my dependencies are back right the bring in if you will. Need this to work on our demo's at least.

**Comment by bradymiller** [comment]
hi @sjpadgett , was off grid for couple days. Code looks good to me and thanks for the fixes/improvements! Also, don't let git beat you down; we can't let git get the best of us :) -brady

**Pull request merged**